librz/core/cdebug.c

```diff
@@ -642,3 +642,116 @@ RZ_API void rz_debug_traces_ascii(RzDebug *dbg, ut64 offset) {
 	rz_table_free(table);
 	rz_list_free(info_list);
 }
+
+/**
+ * \brief Step until end of frame
+ * \param core The RzCore instance
+ * \return success
+ */
+RZ_API bool rz_core_debug_step_until_frame(RzCore *core) {
+	rz_return_val_if_fail(core && core->dbg, false);
+	int maxLoops = 200000;
+	ut64 off, now = rz_debug_reg_get(core->dbg, "SP");
```

Comment:
Is there a way to perform the ESIL operations similarly to what's done below?
```C
if (!rz_core_is_debug(core)) {
		if (!rz_core_esil_step_back(core)) {
			RZ_LOG_ERROR("cannot step back\n");
			return false;
		}
		return true;
	}
```

I see this is done in `rz_core_debug_step_back` and few other functions, but not here.